src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -813,7 +813,10 @@ public function addQuantityInformation(
         $show_price = $this->shouldShowPrice($settings, $product);
         $show_availability = $show_price && $settings->stock_management_enabled;
         $this->product['show_availability'] = $show_availability;
-        $product['quantity_wanted'] = $this->getQuantityWanted();
+
+        if (!isset($product['quantity_wanted'])) {
+            $product['quantity_wanted'] = $this->getQuantityWanted();
```

Comment:
Is this part of the fix or is this an optimization ?

Reply:
Yes, this is part of the fix.

We can also test "cart_quantity" to be sure to be in the shopping cart:
`if (!isset($product['quantity_wanted']) && isset($product['cart_quantity'])) {`
Because in the shopping cart, this is not defined and always returns 1:
`private function getQuantityWanted() { return (int) Tools::getValue('quantity_wanted', 1); }`